### PR TITLE
[ML] DF Analytics _explain API should skip object fields

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Object fields cannot be used as features. At the moment _explain
API includes them and even worse it allows it does not error when
an object field is excluded. This creates the expectation to the
user that all children fields will also be excluded while it's not
the case.

This commit omits object fields from the _explain API and also
adds an error if an object field is included or excluded.

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/explain-dfanalytics.asciidoc [hunk 1]
```diff
@@ -45,9 +45,10 @@ The following explanations are provided:
 
 * which fields are included or not in the analysis and why,
 * how much memory is estimated to be required. The estimate can be used when 
-  deciding the appropriate value for `model_memory_limit` setting later on,
+  deciding the appropriate value for `model_memory_limit` setting later on.
 
-about either an existing {dfanalytics-job} or one that has not been created yet.
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/explain-dfanalytics.asciidoc]: This was stated twice in this paragraph so I took the chance to clean it up here.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java [hunk 2]
```diff
@@ -394,4 +421,8 @@ private void addIncludedFields(ExtractedFields extractedFields, Set<FieldSelecti
     private static boolean isBoolean(Set<String> types) {
         return types.size() == 1 && types.contains(BooleanFieldMapper.CONTENT_TYPE);
     }
+
+    private boolean isObject(Set<String> types) {
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java]: static

**Review by przemekwitek — APPROVED** [review]
LGTM

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Pull request merged**